### DISCUSSION TITLE
egg中进行异常捕获时，能否指定某些错误不被写入日志

### DISCUSSION THREAD

**Discussion by wangyige0701** [eggjs/egg#5310] [Q&A]
我在写接口时因为经常复用逻辑，所以处理异常请求时选择直接在代码里抛出一个错误，然后在最外层的配置里添加config.onerror进行统一捕获并返回错误响应。

但是这样就造成每次抛出异常，相关的错误堆栈信息都会被记录到日志里，但是这些信息并不需要被记录，不知道egg里的错误捕获能不能支持在写入日志前进行一次判断。

**Comment by svenjia** [comment]
不抛出错误，写个中间件拦截异常。

**Reply by wangyige0701** [reply]
感谢
